testprojects/src/java/org/pantsbuild/example/BUILD

```diff
@@ -0,0 +1,16 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+java_library(
+    dependencies=[
+      ":lockfile",
+    ],
+)
+
+coursier_lockfile(
+    name = "lockfile",
+    maven_requirements = [],
+    sources = [
+        "coursier_resolve.lockfile",
+    ],
+)
```

Comment:
Echoing feedback I received on a previous review: is testprojects not now discouraged?

Reply:
I'm still not a big fan because it results in brittle tests, multiple of them depending on the same fixture.

Reply:
There are no tests depending on it. I could move it under `examples`? I find it easier in many cases to iterate on fixing issues outside of the test harness (...should probably invest time in fixing that, but).

Reply:
Yeah I do agree there, I've found it useful for Go too. But then we get into a situation if we don't have tests where we risk bit rot 🤔

Maybe the solution is have them, but don't rely on them for intricate testing like we did in v1?

Reply:
Will leave this here, and have added a test.